### PR TITLE
test(fuzz): add FuzzImportEnvelope for settingsio.Service.Import

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary
- M49 W3 sub-task 3A. Adds `FuzzImportEnvelope` covering `Service.Import` -> `decryptWithPassphrase` -> `json.Unmarshal` -> per-section import.
- 12 seeds (`f.Add(...)`) covering encrypted bundles, unencrypted envelope variants, malformed AES-GCM ciphertexts, oversized/empty/deeply-nested arrays, and version-gating edge cases.
- 6,019 execs/10s on local smoke run.

## Design note
The fuzz body uses a single path that hands the fuzz bytes to `Service.Import` as the encrypted envelope. An earlier draft also had a "Path B" that re-encrypted the fuzz bytes via `encryptWithPassphrase` on every iteration so they would always make it past decrypt to the inner `json.Unmarshal` + per-section import. That path was dropped because `encryptWithPassphrase` runs 600,000 PBKDF2 iterations per call (~770ms), which capped throughput at ~13 execs/10s -- too slow to be useful for the weekly fuzz cron. The seed corpus already includes several valid encrypted payloads that exercise the full decrypt -> Unmarshal -> import path on every fuzz replay; mutations then push outward from there. Production code was deliberately not modified in this PR.

## Test plan
- [x] `go test ./internal/settingsio/...`
- [x] `go test -run=^FuzzImportEnvelope$ -fuzz=^FuzzImportEnvelope$ -fuzztime=10s ./internal/settingsio` -> 6,019 execs, no panics
- [x] `golangci-lint run ./internal/settingsio/...` clean

Closes #1361.

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive fuzz tests for settings import/envelope handling. They exercise encrypted and malformed payloads, boundary and version edge cases, oversized/deep inputs, and invalid ciphertext to ensure the import path remains stable (no panics) across unexpected inputs.

[Review Change Stack](https://app.coderabbit.ai/change-stack/sydlexius/stillwater/pull/1470)

### CODE REVIEW

File: internal/settingsio/export_fuzz_test.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+		APITokens: manyTokens,
+	}))
+
+	// Seed 7: deeply nested values -- a settings map whose values are long
+	// repeated strings to push JSON parsing beyond typical sizes.
+	deepSettings := make(map[string]string, 50)
+	for i := range 50 {
+		deepSettings[strings.Repeat("k", i+1)] = strings.Repeat("v", 1000)
+	}
+	f.Add(encryptPayload(f, Payload{Settings: deepSettings}))
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/settingsio/export_fuzz_test.go]: Fixed in 088d0ffa — corrected the seed comment to describe what the seed actually does (large-value pressure, 50 keys × long string values). Kept the flat `map[string]string`; the proposed `map[string]any` recursive nesting would bypass `Payload`'s `Settings map[string]string` typing and exercise a `json.Unmarshal` error path already covered by seeds 2-4.

File: internal/settingsio/export_fuzz_test.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+		Rules: []RuleExport{
+			{ID: "", Enabled: true, AutomationMode: "auto"},                        // empty ID => skip
+			{ID: "nonexistent_rule_xyz", Enabled: true, AutomationMode: "invalid"}, // bad mode => skip
+		},
+	}))
+
+	// Seed 9: envelope with version "" (treated as "1.0" for backward compat).
+	f.Add(encryptPayload(f, map[string]any{
+		"settings": map[string]string{"legacy": "yes"},
+	}))
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/settingsio/export_fuzz_test.go]: Fixed in 088d0ffa — Seed 9 now constructs the `Envelope` inline with explicit `Version: ""` so the legacy backward-compat branch in `Import` is actually exercised. The one-time `encryptWithPassphrase` call in seed setup is fine — not per-iteration (that was the optimization-round-2 concern).

**Comment by sydlexius** [comment]
@coderabbitai resolve
